manifests/init.pp

```diff
@@ -116,63 +122,105 @@
+  Hash                          $default_cfg,
+  Hash                          $cfg,
+  Hash                          $ldap_cfg,
+  String                        $install_method,
+  Boolean                       $use_internet_repo,
+  String                        $rpm_iteration,
+  Boolean                       $simp_dashboards,
+  String                        $app_pki_dir,
+  Hash                          $simp_ldap_conf,
+  String                        $admin_pw                = simplib::passgen('simp_grafana'),
+  Simplib::Netlist              $trusted_nets            = simplib::lookup('simp_options::trusted_nets', { 'default_value' => ['127.0.0.1/8'] }),
+  Boolean                       $firewall                = simplib::lookup('simp_options::firewall', { 'default_value' => false }),
+  Boolean                       $ldap                    = simplib::lookup('simp_options::ldap', { 'default_value' => false }),
+  String                        $version                 = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' }),
   Variant[Boolean,Enum['simp']] $pki                     = simplib::lookup('simp_options::pki', { 'default_value' => false }),
   String                        $app_pki_external_source = simplib::lookup('simp_options::pki::source', { 'default_value' => '/etc/pki/simp/x509' }),
-  Hash                          $cfg                     = {},
-  Hash                          $ldap_cfg                = {},
-  String                        $install_method          = 'repo',
-  Boolean                       $use_internet_repo       = false,
-  String                        $version                 = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' }),
-  String                        $rpm_iteration           = '1',
-  Boolean                       $simp_dashboards         = false
-) inherits ::simp_grafana::params {
+  String                        $app_pki_key             = "${app_pki_dir}/private/${facts['fqdn']}.pem",
+  String                        $app_pki_cert            = "${app_pki_dir}/public/${facts['fqdn']}.pub",
+  String                        $base_dn                 = simplib::lookup('simp_options::ldap::base_dn', { 'default_value' => simplib::ldap::domain_to_dn("example.com") } ),
+  String                        $bind_dn                 = simplib::lookup('simp_options::ldap::bind_dn', { 'default_value' => "uid=%s,${base_dn}" } ),
+  String                        $bind_pw                 = simplib::lookup('simp_options::ldap::bind_pw', { 'default_value' => undef } ),
+  Array[Simplib::URI,1]         $ldap_urls               = simplib::lookup('simp_options::ldap::uri', { 'default_value' => undef } )
+) {
 
-  $merged_cfg = deep_merge($::simp_grafana::params::cfg, $cfg)
-  $merged_ldap_cfg = deep_merge($::simp_grafana::params::ldap_cfg, $ldap_cfg)
+  $_simp_ldap_server_name = split($ldap_urls[0], 'ldap://')[1]
```

Comment:
What about ldaps://

Reply:
Grafana does not use a protocol at all, so I only need to extract the hostname. I will update it to be more flexible. It also only allows configuration of one LDAP server.